Sources/SnapshotTesting/Common/ForwardingWKNavigationDelegate.swift

```diff
@@ -0,0 +1,133 @@
+#if os(macOS)
+import Cocoa
+#endif
+#if os(iOS)
+import UIKit
+#endif
+
+#if os(iOS) || os(macOS)
+import WebKit
+
+final class ForwardingWKNavigationDelegate: NSObject {
+  var didFinish: (() -> Void)?
+  var original: WKNavigationDelegate?
+  init(didFinish: @escaping () -> Void = {}, originalDelegate: WKNavigationDelegate?) {
+    self.original = originalDelegate
+    self.didFinish = didFinish
+  }
+
+  override public func responds(to aSelector: Selector!) -> Bool {
+    if decidePolicyPre13 == aSelector, original?.responds(to: aSelector) != true {
+      return NSObject.responds(to: aSelector)
+    }
+
+    if
+      #available(iOSApplicationExtension 13.0, *, OSXApplicationExtension 10.15, *),
+      decidePolicy13 == aSelector,
+      original?.responds(to: aSelector) != true
+    {
+      return NSObject.responds(to: aSelector)
+    }
+
+    return interceptedSelectors.contains(aSelector)
+      || original?.responds(to: aSelector) == true
+      || NSObject.responds(to: aSelector)
+  }
```

Comment:
@Sherlouk this works great. The only thing is that there are 2 optional delegate methods that might be implemented by the user:

```
func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, preferences: WKWebpagePreferences, decisionHandler: @escaping (WKNavigationActionPolicy, WKWebpagePreferences) -> Void)

func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void)
```

If the user does not choose to implement them, we should not respond to the methods, otherwise we need to. See below. 

This code is not pretty, but handeling this in the methods instead of here is even worse.